prow/resultstore/writer/writer.go

```diff
@@ -70,13 +73,25 @@ type writer struct {
 	finalized   bool
 }
 
+func retryable(err error) bool {
+	status, _ := status.FromError(err)
+	switch status.Code() {
+	case codes.Aborted:
+	case codes.DeadlineExceeded:
+	case codes.Unavailable:
```

Comment:
Could you explain briefly in the comments why these conditions should not be retried? I ask because my assumption is that we should default to retrying (e.g., if there are network issues, we want to retry because those issues may be transient), so categorically disabling retries for certain conditions appears overly conservative to me.

Reply:
This switch statement looks reversed.

These are the ones that should be retried.

Reply:
Outdated, but note that Go fall-through semantics say that the default applies unless a case above is matched. I believe it was written correctly, but I've refactored after discussion with ashleyrichter who easily persuaded me a list of persistent errors is a better approach, and the list of persistent error codes they use in clients.